### PR TITLE
Player coloring and game over handling

### LINKED ISSUE #29: GD: Add title screen and add results to lobby

### PR DESCRIPTION
**Pull request by mineiwik**
Closes #29

### CODE REVIEW

File: scripts/controller.gd [hunk 1]
```diff
@@ -78,18 +81,24 @@ func _start_round(payload: Dictionary):
...
 		var piece_pos = Position.get_obj_from_dict(piece["position"])
 		self._board.place_piece(piece["piece_id"], piece["player_id"], piece_pos.get_vec())
-	self._board.turn_all_player_piece_lights_on(self._selectable_light_intensity)
 
 func _animate_round(payload: Dictionary):
 	await self._board.animate_events(payload["timeline"])
+	if not payload["game_over"] == null:
+		GlobalVariables.winner_id = payload["game_over"]["winner_player_id"]
+		get_tree().change_scene_to_file("res://scenes/lobby.tscn")
+	self._current_state = STATES.AWAITING_ROUND
```
Comment (siku2) [inline scripts/controller.gd]: The client still needs to send the ready_for_next_round message here. See https://github.com/dsg-jam/ld51-server/blob/main/docs/protocol.md#game-over

File: scripts/controller.gd [hunk 2]
```diff
@@ -64,12 +65,14 @@ func _on_ws_received_message(_msg_type: String) -> void:
...
 				if self._current_state == STATES.AWAITING_ROUND:
-					self._start_round(msg["payload"])
 					self._current_state = STATES.ONGOING_ROUND
+					self._start_round(msg["payload"])
 			DSGMessageType.ROUND_RESULT:
 				if self._current_state == STATES.AWAITING_RESULT:
-					self._animate_round(msg["payload"])
 					self._current_state = STATES.ONGOING_ANIMATION
+					self._animate_round(msg["payload"])
+			DSGMessageType.ERROR:
```
Comment (siku2) [inline scripts/controller.gd]: The `MSG_FILTER` currently makes this unreachable. I think we should catch and log the error in `DSGNetwork` to make sure they're always logged, regardless of the state.

File: scripts/lobby.gd [hunk 3]
```diff
@@ -55,18 +68,29 @@ func _on_ws_received_message(_msg_type: String) -> void:
...
 
 func _server_hello(payload: Variant):
 	if not payload["is_host"]:
 		self._setup_non_host()
-	GlobalVariables.player_id = payload["player_id"]
+	var player = payload["player"]
+	GlobalVariables.player_id = player["id"]
+	GlobalVariables.player_number = player["number"]
+	GlobalVariables.session_id = payload["session_id"]
+	self._info.text = "Welcome! You play %s." % GlobalVariables.COLOR_MAPPING[GlobalVariables.player_number]
```
Comment (siku2) [inline scripts/lobby.gd:81]: The `server_hello` message now also contains the list of all players that are already in the lobby.

See:
https://github.com/dsg-jam/ld51-server/blob/ad72f6c3a7a7d7507486315609c0f5ced8f6c18c/ld51_server/protocol/lobby.py#L16

We should at least use this to update the `amount_of_players`.

Comment (mineiwik) [reply scripts/lobby.gd:81]: 18a9ee2

File: scripts/controller.gd (hunk 2 above)
Comment (mineiwik) [reply scripts/controller.gd]: 18a9ee2 - errors are currently not logged (todo)

File: scripts/controller.gd (hunk 1 above)
Comment (mineiwik) [reply scripts/controller.gd]: 18a9ee2

**Review by siku2 — APPROVED** [review]

**Pull request merged**